actionpack/lib/action_dispatch/railtie.rb

```diff
@@ -13,12 +13,12 @@ class Railtie < Rails::Railtie # :nodoc:
     config.action_dispatch.default_charset = nil
     config.action_dispatch.rack_cache = false
     config.action_dispatch.http_auth_salt = "http authentication"
-    config.action_dispatch.signed_cookie_salt = "signed cookie"
-    config.action_dispatch.encrypted_cookie_salt = "encrypted cookie"
-    config.action_dispatch.encrypted_signed_cookie_salt = "signed encrypted cookie"
     config.action_dispatch.use_authenticated_cookie_encryption = false
     config.action_dispatch.perform_deep_munge = true
 
+    config.action_dispatch.legacy_encryptor_schemes = []
+    config.action_dispatch.legacy_verifier_schemes  = []
```

Comment:
I find the `scheme` wording confusing. Isn't that what `aes-256-gcm` is? Also dislike the `legacy` wording since it doesn't clarify how people can use this to rotate their keys.

Ideally I'd like the config to be something like:

```ruby
# config/initializers/rotations.rb
# Here's some instructions on how to do cookie rotations right and why should be.
#   - Rotate your secret key base whenever a coworker leaves, etc.
Rails.application.config.secrets.tap do |secrets|
  secrets.rotate_out secret_key_base: secrets.old_secret_key_base
  secrets.rotate_out cipher: 'aes-256-gcm'
  secrets.rotate_out :signed, digest: 'xxx' # We need some way to explain that a rotation targets either signed or encrypted messages.
end

Rails.application.config.action_dispatch.tap do |config|
  config.cookies.rotate_out signed_salt: config.old_signed_salt # It'll use the new config.signed_salt but add this as a fallback.
end
```

With that API our new framework defaults could then contain:

```ruby
# config/initializers/new_framework_defaults_5_2.rb
# …

# Use AES 256 GCM authenticated encryption in cookies and ActiveSupport::MessageEncryptor.
# Rails.application.secrets.rotate_out cipher: 'aes-xx-cbc'
```